From ff4d26816ded3da87c393b1accd07a63ee8a91cb Mon Sep 17 00:00:00 2001
From: Julien Jorge <julien.jorge@stuff-o-matic.com>
Date: Fri, 25 Oct 2013 22:21:35 +0200
Subject: [PATCH] Force static libraries to be compiled as static libraries.

There were some add_library()
instructions in the CMakeLists.txt which did not have the STATIC/SHARED
keywords. I expected the default value to be STATIC but it depends upon
the value of BUILD_SHARED_LIBS and BUILD_STATIC_LIBS.
---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b907e6..830d60c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,7 +76,7 @@ set( CLAW_APPLICATION_SOURCES
   ${CLAW_CODE_DIR}arguments_table.cpp
 )
 add_library( claw_application SHARED ${CLAW_APPLICATION_SOURCES} )
-add_library( claw_application-static ${CLAW_APPLICATION_SOURCES} )
+add_library( claw_application-static STATIC ${CLAW_APPLICATION_SOURCES} )
 target_link_libraries( claw_application claw_logger )
 
 if( NOT WIN32 )
@@ -98,7 +98,7 @@ set( CLAW_CONFIGURATION_FILE_SOURCES
 )
 add_library( claw_configuration_file SHARED ${CLAW_CONFIGURATION_FILE_SOURCES} )
 add_library(
-  claw_configuration_file-static
+  claw_configuration_file-static STATIC
   STATIC ${CLAW_CONFIGURATION_FILE_SOURCES}
 )
 
@@ -108,7 +108,7 @@ set( CLAW_DYNAMIC_LIBRARY_SOURCES
 )
 add_library( claw_dynamic_library SHARED ${CLAW_DYNAMIC_LIBRARY_SOURCES} )
 add_library(
-  claw_dynamic_library-static
+  claw_dynamic_library-static STATIC
   STATIC ${CLAW_DYNAMIC_LIBRARY_SOURCES}
 )
 target_link_libraries( claw_dynamic_library ${CMAKE_DL_LIBS} )
-- 
2.26.2

